src/units/unit.cpp

```diff
@@ -1395,12 +1395,132 @@ bool unit::has_ability_by_id(const std::string& ability) const
 	return false;
 }
 
-void unit::remove_ability_by_id(const std::string& ability)
+static bool matches_ability_filter(config::all_children_iterator i, const config & filter, const std::string& ability, const std::set<std::string> checking_tags, bool in_tag)
+{
+	const std::string& ab_active_on = !i->cfg["active_on"].empty() ? i->cfg["active_on"].str() : "both";
+	const std::string& ab_overwrite_specials = !i->cfg["overwrite_specials"].empty() ? i->cfg["overwrite_specials"].str() : "none";
+	const std::string& ab_apply_to = !i->cfg["apply_to"].empty() ? i->cfg["apply_to"].str() : "self";
+	bool ab_affect_self = i->cfg["affect_self"].to_bool(true);
+	bool ab_affect_allies = i->cfg["affect_allies"].to_bool(true);
+	bool ab_affect_enemies = i->cfg["affect_enemies"].to_bool();
+	bool ab_cumulative = i->cfg["cumulative"].to_bool();
```

Comment:
I'm not sure which existing C++ functions would be the right ones to use, but a lot of this feels like code that should already have convenience functions in the existing filter implementations.

Reply:
true for affect_self who have an function callable from unit::: function but not other in namespace in abilities.cpp

Reply:
it not for verify what affect_allies/enemies were apppied to unit adjacent but only what attribute present in abilities with [affect_adjacent] tags